python_modules/libraries/dagster-pyspark/dagster_pyspark/types.py

```diff
@@ -13,17 +23,476 @@
+                    String, is_required=False, description="names of partitioning columns."
+                ),
+                'compression': Field(
+                    Enum(
+                        'compression',
+                        [
+                            EnumValue('none'),
+                            EnumValue('uncompressed'),
+                            EnumValue('snappy'),
+                            EnumValue('gzip'),
+                            EnumValue('lzo'),
+                            EnumValue('brotli'),
+                            EnumValue('lz4'),
+                            EnumValue('zstd'),
+                        ],
+                    ),
+                    is_required=False,
+                    description="compression codec to use when saving to file. This will override ``spark.sql.parquet.compression.codec``. If None is set, it uses the value specified in ``spark.sql.parquet.compression.codec``.",
+                ),
+            },
+            'json': {
+                'path': Field(
+                    String,
+                    is_required=True,
+                    description="the path in any Hadoop supported file system.",
+                ),
+                'mode': Field(
+                    Enum(
+                        'mode',
+                        [
+                            EnumValue(
+                                'append',
+                                description="Append contents of this :class:`DataFrame` to existing data.",
+                            ),
+                            EnumValue('overwrite', description="Overwrite existing data."),
+                            EnumValue(
+                                'ignore',
+                                description="Silently ignore this operation if data already exists.",
+                            ),
+                            EnumValue(
+                                'error',
+                                description="(default case): Throw an exception if data already exists.",
+                            ),
+                            EnumValue(
+                                'errorifexists',
+                                description="(default case): Throw an exception if data already exists.",
+                            ),
+                        ],
+                    ),
+                    is_required=False,
+                    description="specifies the behavior of the save operation when data already exists.",
+                ),
+                'compression': Field(
+                    Enum(
+                        'compression',
+                        [
+                            EnumValue('none'),
+                            EnumValue('bzip2'),
+                            EnumValue('gzip'),
+                            EnumValue('lz4'),
+                            EnumValue('snappy'),
+                            EnumValue('deflate'),
+                        ],
```

Comment:
same feedback about this enum

Reply:
'compression' `Enum` is not the same for all types.